eth_utils/abi.py

```diff
@@ -1,29 +1,70 @@
+from collections import (
+    abc,
+)
+import copy
+import itertools
+import re
 from typing import (
     Any,
     Dict,
+    Iterable,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+    cast,
+)
+
+from eth_typing import (
+    ABI,
+    ABIComponent,
+    ABIElement,
+    ABIError,
+    ABIEvent,
+    ABIFallback,
+    ABIFunction,
+    ABIReceive,
+)
+from typing_extensions import (
+    deprecated,
+)
+
+from eth_utils.types import (
+    is_list_like,
 )
 
 from .crypto import (
     keccak,
 )
 
 
+@deprecated(
```

Comment:
How are we releasing this? Since this has breaking changes I think we may want to avoid any deprecation here if we're going to do a new major release and choose to deprecate in the current running version of `eth_utils`. I made a similar comment in the `v7` web3.py PR about removing things slated for `main` and deprecating if `v6` there. 

Thoughts on that?

Reply:
Yeah agreed this will be removed now that we're releasing a major.